sycl/include/sycl/ext/oneapi/experimental/profiling_tag.hpp

```diff
@@ -0,0 +1,44 @@
+//==--------- profiling_tag.hpp --- SYCL profiling tag extension -----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/aspects.hpp>
+#include <sycl/event.hpp>
+#include <sycl/handler.hpp>
+#include <sycl/properties/queue_properties.hpp>
+#include <sycl/queue.hpp>
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+
+inline event submit_profiling_tag(queue &Queue,
+                                  const sycl::detail::code_location &CodeLoc =
+                                      sycl::detail::code_location::current()) {
+  if (Queue.get_device().has(aspect::ext_oneapi_queue_profiling_tag)) {
+    return Queue.submit(
+        [=](handler &CGH) {
+          sycl::detail::HandlerAccess::internalProfilingTagImpl(CGH);
+        },
+        CodeLoc);
+  }
+
+  // If it is not supported natively on the device, we use another path if
+  // profiling is enabled.
+  if (!Queue.has_property<sycl::property::queue::enable_profiling>())
```

Comment:
Is this an `else if`? From the error message wording it looks like you want one or the other